src/Datasource/InvalidPropertyInterface.php

```diff
@@ -17,6 +17,9 @@
 /**
  * Describes the methods that any class representing a data storage should
  * comply with.
+ *
+ * @method \Cake\Datasource\EntityTrait getInvalid()
```

Comment:
@cleptric you just type hint return types as regular.

```php
/**
 * @param string $foo
 * @return \Bar
 */
public function foo($bar);
```
Is same as:
```php
/**
 * @method \Bar foo(string $bar)
 */
```
See: https://phpdoc.org/docs/latest/references/phpdoc/tags/method.html